code/game/objects/structures/window.dm

```diff
@@ -250,46 +244,98 @@
+			// at most 60 without armor , 23 with 15 melle armor
+			target.damage_through_armor(round(20 * skillRatio * health/maxhealth / (toughTarget ? 3 : 1)), BRUTE, BP_HEAD, ARMOR_MELEE, sharp = FALSE, armor_divisor = 0.4)
+			hit(round(target.mob_size * skillRatio * 1.5 * (toughTarget ? 2 : 1) / windowResistance))
 		if(GRAB_NECK)
 			visible_message(SPAN_DANGER("<big>[user] crushes [target] against \the [src]!</big>"))
-			target.Weaken(5)
-			target.damage_through_armor(12, BRUTE, BP_HEAD, ARMOR_MELEE)
-			hit(20)
+			// at most 90 damage without armor, 40 with 15 melle armor
+			target.damage_through_armor(round(30 * skillRatio * health/maxhealth / (toughTarget ? 3 : 1)), BRUTE, BP_HEAD, ARMOR_MELEE, sharp = FALSE, armor_divisor = 0.3)
+			target.stats.addTempStat(STAT_VIG, -STAT_LEVEL_ADEPT * 2, toughTarget ? 10 SECONDS : 20 SECONDS, "window_smash")
+			hit(round(target.mob_size * skillRatio * 2 * ((toughTarget ? 2 : 1)) / windowResistance))
 	admin_attack_log(user, target,
 		"Smashed [key_name(target)] against \the [src]",
 		"Smashed against \the [src] by [key_name(user)]",
 		"smashed [key_name(target)] against \the [src]."
 	)
-	sleep(5) //Allow a littleanimating time
+	end_grab_onto(user, target)
 	return TRUE
 
-/obj/structure/window/proc/hit_by_living(var/mob/living/M)
+proc/end_grab_onto(mob/living/user, mob/living/target)
+	for(var/obj/item/grab/G in list(user.l_hand, user.r_hand))
+		if(G.affecting == target)
+			qdel(G)
+			break
+
+/obj/structure/window/proc/hit_by_living(mob/living/M)
 	var/body_part = pick(BP_HEAD, BP_CHEST, BP_GROIN)
 	var/direction = get_dir(M, src)
+	var/tforce = M.mob_size
 	visible_message(SPAN_DANGER("[M] slams against \the [src]!"))
-	if(prob(30))
-		M.Weaken(1)
-	M.damage_through_armor(rand(7,10), BRUTE, body_part, ARMOR_MELEE)
+	// being super tough has its perks!
+	if(!M.stats.getPerk(PERK_ASS_OF_CONCRETE))
+		var/victimToughness = M.stats.getStat(STAT_TGH, FALSE)
+		victimToughness = victimToughness ? victimToughness : 1
+		var/windowResistance = resistance ? resistance : 1
+		var/healthRatio = health/maxhealth
+		// you shall suffer for being negative on toughness , it becomes negative so it cancels the negative toughness
+		var/toughnessDivisor = victimToughness > 0 ? 100 : -(100 - victimToughness)
+		// if you less tougher and less sized than the window itself and its health , you are more likely to suffer more
+		if(victimToughness * M.mob_size / toughnessDivisor < windowResistance * healthRatio)
+			M.adjustHalLoss(5)
+			M.Weaken(2)
+			// 40 in worst case, 10 with 15 melle armor
+			M.damage_through_armor(40 * (1 - victimToughness/toughnessDivisor) * healthRatio, BRUTE, body_part, ARMOR_MELEE, sharp = FALSE, armor_divisor = 0.5)
+		else
+			M.adjustHalLoss(3)
+			// 20 in worst  case , 5 with 15 melle armor
+			M.damage_through_armor(20 * (1 - victimToughness/toughnessDivisor) * healthRatio, BRUTE, body_part, ARMOR_MELEE, sharp = FALSE)
```

Comment:
If the victim has more than 100 thoughness (the current max is 150) then they will take negative damage because of `(1 - victimToughness/toughnessDivisor)` since `thoughnessDiviser = 100`